### ISSUE TITLE
Hello World fails on ArchLinux - Docker 1.10.2

### ISSUE THREAD

**Issue by Klaim** [docker/docker#20653]
Output of `docker version`:

```
╰─$ docker version
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.6
 Git commit:   c3959b1
 Built:        Tue Feb 23 00:53:12 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.6
 Git commit:   c3959b1
 Built:        Tue Feb 23 00:53:12 2016
 OS/Arch:      linux/amd64
```

Output of `docker info`:

```
╰─$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.10.2
Storage Driver: devicemapper
 Pool Name: docker-8:2-6555021-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 10.94 MB
 Data Space Total: 107.4 GB
 Data Space Available: 107.4 GB
 Metadata Space Used: 581.6 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpo
oldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.115 (2016-01-25)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 3.14.32-xxxx-grs-ipv6-64
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.72 GiB
Name: klaimslab
ID: AXV2:ABSB:5LAB:E2RA:JRHD:MNHG:3E6G:KP6W:HNST:2XB2:QQRJ:PHKD
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
```

Provide additional environment details (AWS, VirtualBox, physical, etc.):

List the steps to reproduce the issue:
1. Install Docker through pacman: "pacman -S docker"
2. Run "docker run hello-world"

Describe the results you received:

```
╰─$ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world

03f4658f8b78: Extracting [==================================================>]    601 B/601 B
a3ed95caeb02: Download complete
docker: failed to register layer: devmapper: Error mounting '/dev/mapper/docker-8:2-6555021-f67bdeaadec34a41c7a3e2d
fc70e6cad1b15832966fd060c7f57295dcd6ecd98' on '/var/lib/docker/devicemapper/mnt/f67bdeaadec34a41c7a3e2dfc70e6cad1b1
5832966fd060c7f57295dcd6ecd98': invalid argument.
See 'docker run --help'.
```

Describe the results you expected:

No error.

Provide additional info you think is important:

I discovered this issue while trying to install Discourse (http://discourse.org) which fails because of the very specific same error.

I tried uninstalling completely docker then re-installing it with no change to this output.

**Comment by thaJeztah** [comment]
do the daemon logs show anything useful?

I noticed that the binary is built using Go 1.6; since the Arch linux version is not packaged by Docker, have you tried reporting an issue there?

**Comment by cpuguy83** [comment]
Yes, go 1.6 is not supported yet as there are known issues in our code before it can be used with 1.6.

**Comment by thaJeztah** [comment]
@Klaim since this can still be an issue in Docker itself, can you

- check if the system and daemon logs show anything useful?
- tell if this was a fresh install, or an update from an older version?
- reporting at the team packaging this version won't hurt; as mentioned we don't build (or test) on Arch linux, so any help would be appreciated

**Comment by Klaim** [comment]
Thanks for the quick answers.

1. I am not a linux expert so I'm not totally sure where to check for these logs exactly.
Here are some reports, but I see nothing wrong:

```
╰─$ systemctl status docker
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: active (running) since Wed 2016-02-24 01:48:00 CET; 21h ago
     Docs: https://docs.docker.com
 Main PID: 3217 (docker)
   CGroup: /system.slice/docker.service
           └─3217 /usr/bin/docker daemon -H fd://
```
(last time I installed docker to try it again from fresh install was indeed last night).

I tried the following steps:
 a. restart docker daemon : `systemctl restart docker`
 b. run hello world
 c. extract docker's log using `journalctl -u docker`

Here is the result (starting from the restart):

```
Feb 24 23:32:18 klaimslab docker[8460]: time="2016-02-24T23:32:18.919816320+01:00" level=info msg="Processing signal 'terminated'"
Feb 24 23:32:19 klaimslab systemd[1]: Stopped Docker Application Container Engine.
Feb 24 23:32:19 klaimslab systemd[1]: Starting Docker Application Container Engine...
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.149533079+01:00" level=warning msg="devmapper: Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man docker` to refer to dm.thinpooldev section."
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.159983033+01:00" level=warning msg="devmapper: Base device already exists and has filesystem xfs on it. User specified filesystem  will be ignored."
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.163708197+01:00" level=info msg="[graphdriver] using prior storage driver \"devicemapper\""
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.164307795+01:00" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.166322537+01:00" level=warning msg="Running modprobe nf_nat failed with message: `modprobe: WARNING: Module nf_nat not found in directory /lib/modules/3.14.32-xxxx-grs-ipv6-64`, error: exit status 1"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.169309681+01:00" level=info msg="Firewalld running: false"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.550442313+01:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.827455874+01:00" level=warning msg="Your kernel does not support kernel memory limit."
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.827622569+01:00" level=warning msg="Your kernel does not support cgroup cfs period"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.827653346+01:00" level=warning msg="Your kernel does not support cgroup cfs quotas"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.828670594+01:00" level=info msg="Loading containers: start."
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.828748459+01:00" level=info msg="Loading containers: done."
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.828769282+01:00" level=info msg="Daemon has completed initialization"
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.828796499+01:00" level=info msg="Docker daemon" commit=c3959b1 execdriver=native-0.2 graphdriver=devicemapper version=1.10.2
Feb 24 23:32:19 klaimslab systemd[1]: Started Docker Application Container Engine.
Feb 24 23:32:19 klaimslab docker[8717]: time="2016-02-24T23:32:19.838592977+01:00" level=info msg="API listen on /var/run/docker.sock"
Feb 24 23:32:39 klaimslab docker[8717]: time="2016-02-24T23:32:39.432368584+01:00" level=error msg="Handler for POST /v1.22/containers/create returned error: No such image: hello-world:latest"
```

I'm guessing that this last error is a side effect of the initial error log in CLI, but I might be wrong.

I don't see anything else, `journalctl`  don't display anything related.

2. It was a fresh install yes, I only installed a few things before docker: HHVM, NGinx and PostGreSQL.
As said before, I also tried to uninstall docker completely (using your documentation for archlinux) and reinstalled it and run hello-world immediately after which lead to no success so far.

3. I will try to report it, though I'm not sure where yet (certainly their forum). Will keep you up to date if something comes up.

**Comment by endophage** [comment]
I've been having the same issue with 1.10.2. Dug the following out of the logs:

With the RC (on push):
```
ERRO[0010] Upload failed, retrying: devmapper: Unknown device 8a648f689ddb6408f71f7a41025b56583f5a9a0c5fb6df105d97c8e0ea139910 
ERRO[0015] Upload failed, retrying: devmapper: Unknown device 8a648f689ddb6408f71f7a41025b56583f5a9a0c5fb6df105d97c8e0ea139910 
ERRO[0041] Upload failed, retrying: net/http: TLS handshake timeout 
ERRO[0056] Upload failed, retrying: devmapper: Unknown device 8a648f689ddb6408f71f7a41025b56583f5a9a0c5fb6df105d97c8e0ea139910 
ERRO[0078] Upload failed: devmapper: Unknown device 8a648f689ddb6408f71f7a41025b56583f5a9a0c5fb6df105d97c8e0ea139910
```

With the final release (on pull):
```
Feb 24 16:21:59 endophage systemd[1080]: dev-disk-by\x2duuid-a5c84592\x2d3633\x2d40b1\x2d868d\x2d2c282b2be425.device: Dev dev-disk-by\x2duuid-a5c84592\x2d3633\x2d40b1\x2d868d\x2d2c282b2be425.device appeared twice with different sysfs paths /sys/devices/virtual/block/loop0 and /sys/devices/virtual/block/dm-5
```

**Comment by thaJeztah** [comment]
@endophage opened an issue for docker being built with Go 1.6 (yay), and it looks like they released a new version; https://bugs.archlinux.org/task/48329

Interested to hear if that improves the situation!

**Comment by endophage** [comment]
It was fixed very quickly last night. @Klaim please try running `pacman -Syu` and you should get an update to the Arch package `docker-1:1.10.2-2`, which if you run `docker version` will show it was built with Go 1.5.3. I haven't had any issues since installing that version.

**Comment by thaJeztah** [comment]
Thanks for reporting back @endophage 

@Klaim I'm going to go ahead and close this issue, but please let us know if it is resolved for you, and feel free to continue the discussion here

**Comment by Klaim** [comment]
@endophage @thaJeztah The issue is not fixed after upgrading to docker 1.10.2 (did pacman -Syu and verified I got the new version).
The docker version correctly shows it will be buit with Go 1.5.3. 

Here is the full log (I tried to uninstall all dependencies and reinstall before trying again):

```
╭─klaim@klaimslab  ~
╰─$ sudo pacman -Rns docker
checking dependencies...

Packages (2) bridge-utils-1.5-2  docker-1:1.10.2-2

Total Removed Size:  40.62 MiB

:: Do you want to remove these packages? [Y/n] y
:: Processing package changes...
(1/2) removing docker                                                            [----------------------------------------------]   0%(1/2) removing docker                                                            [----------------------------------------------]   0%(1/2) removing docker                                                            [##############################################] 100%
(2/2) removing bridge-utils                                                      [----------------------------------------------]   0%(2/2) removing bridge-utils                                                      [----------------------------------------------]   0%(2/2) removing bridge-utils                                                      [##############################################] 100%
╭─klaim@klaimslab  ~
╰─$ sudo rm -rf /var/lib/docker
╭─klaim@klaimslab  ~
╰─$ sudo pacman -Syu docker
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (2) bridge-utils-1.5-2  docker-1:1.10.2-2

Total Installed Size:  40.62 MiB

:: Proceed with installation? [Y/n] y
(0/2) checking keys in keyring                                                   [----------------------------------------------]   0%(1/2) checking keys in keyring                                                   [#######################-----------------------]  50%(2/2) checking keys in keyring                                                   [##############################################] 100%
(0/2) checking package integrity                                                 [----------------------------------------------]   0%(1/2) checking package integrity                                                 [----------------------------------------------]   0%(2/2) checking package integrity                                                 [##############################################] 100%
(0/2) loading package files                                                      [----------------------------------------------]   0%(1/2) loading package files                                                      [----------------------------------------------]   0%(2/2) loading package files                                                      [##############################################] 100%
(0/2) checking for file conflicts                                                [----------------------------------------------]   0%(1/2) checking for file conflicts                                                [#######################-----------------------]  50%(2/2) checking for file conflicts                                                [##############################################] 100%
(0/2) checking available disk space                                              [----------------------------------------------]   0%(1/2) checking available disk space                                              [#######################-----------------------]  50%(2/2) checking available disk space                                              [##############################################] 100%
:: Processing package changes...
(1/2) installing bridge-utils                                                    [----------------------------------------------]   0%(1/2) installing bridge-utils                                                    [##############################################] 100%
(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [----------------------------------------------]   0%(2/2) installing docker                                                          [###########-----------------------------------]  25%(2/2) installing docker                                                          [##############################################] 100%
Optional dependencies for docker
    btrfs-progs: btrfs backend support [installed]
    lxc: lxc backend support
╭─klaim@klaimslab  ~
╰─$ docker version
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Thu Feb 25 09:29:05 2016
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
╭─klaim@klaimslab  ~
╰─$ sudo docker run hello-world                                                                                                   1 ↵
docker: Cannot connect to the Docker daemon. Is the docker daemon running on this host?.
See 'docker run --help'.
╭─klaim@klaimslab  ~
╰─$ sudo systemctl enable docker                                                                                                125 ↵
Created symlink from /etc/systemd/system/multi-user.target.wants/docker.service to /usr/lib/systemd/system/docker.service.
╭─klaim@klaimslab  ~
╰─$ sudo systemctl start docker
╭─klaim@klaimslab  ~
╰─$ sudo docker  run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
03f4658f8b78: Extracting [==================================================>]    601 B/601 B
a3ed95caeb02: Download complete
docker: failed to register layer: devmapper: Error mounting '/dev/mapper/docker-8:2-6555017-e0eb2b6beddfde460e37db285631b200e7554b19f92f6f6516a44213a845219b' on '/var/lib/docker/devicemapper/mnt/e0eb2b6beddfde460e37db285631b200e7554b19f92f6f6516a44213a845219b': invalid argument.
See 'docker run --help'.

```

I also followed the experiments suggested by ArchLinux forum people: https://bbs.archlinux.org/viewtopic.php?pid=1607584

No progress so far except that we know changing devicemapper to overlayfs does not seem to work.

So this issue is still present.

**Comment by thaJeztah** [comment]
Reopening :cry: thanks for testing

**Comment by Klaim** [comment]
Sorry about that :)

Do you have access to a fresh ArchLinux install/vm to check this? It seems very simple to test.

**Comment by endophage** [comment]
:disappointed: my logs did show a different error so not too surprising it wasn't a universal fix.

**Comment by Klaim** [comment]
OK I fixed it with help from the ArchLinux people (see the thread: https://bbs.archlinux.org/viewtopic.php?pid=1607584 )

To summarize:

1. We realized that something was wrong with my kernel image (missing stuffs);
2. I found that indeed the image was a custom one setup by my server provider (OVH/kimsuffi);
3. I had to learn how to switch kernel (I installed linux-grsec) using pacman and grub;
4. After finally switching kernel, I run the hello world with success:

```
╰─$ sudo docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
03f4658f8b78: Pull complete
a3ed95caeb02: Pull complete
Digest: sha256:8be990ef2aeb16dbcb9271ddfe2610fa6658d13f6dfb8bc72074cc1ca36966a7
Status: Downloaded newer image for hello-world:latest

Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker Hub account:
 https://hub.docker.com

For more examples and ideas, visit:
 https://docs.docker.com/userguide/
````

So basically: kernel version was wrong. 

Would it be possible to add a check in docker (maybe in packages) to avoid this issue in the future?

Thanks for your time.

**Comment by thaJeztah** [comment]
Just read that thread you linked to; I don't understand why in one of your outputs, the daemon started with `-s overlay`, but later on it refused to start (as expected), because overlay was not found?

```
sudo docker daemon -s overlay                                                                                                 1 ↵
ERRO[0000] 'overlay' not found as a supported filesystem on this host. Please ensure kernel is new enough and has overlay support loaded.
FATA[0000] Error starting daemon: error initializing graphdriver: driver not supported
```

**Comment by Klaim** [comment]
The guy on the thread first said

**Comment by jesusvazquez** [comment]
Hello @Klaim same issue in my kimsufi! I'm going to try to switch the kernel version.

Thank you!

**Comment by jesusvazquez** [comment]
Hello Again!, 

I had a few minutes this sunday and I tried to fix this issue. Here is a quick solution: 

First you need to check the kernel your system is using. This issue is probably only related with kimsufi archlinux image. 

```
# ls -1 /boot                                                                                                                                
System.map-3.14.32-xxxx-grs-ipv6-64
bzImage-3.14.32-xxxx-grs-ipv6-64
grub
initramfs-linux-lts-fallback.img
initramfs-linux-lts.img
intel-ucode.img
vmlinuz-linux-lts
```
Their custom kernel is **linux-3.14.32-xxxx-grs-ipv6-64** and I decided to install the **lts** version in my server because I will use it only as a host to perform testing inside docker containers. That explain the **vmlinuz-linux-lts** image. You can also install the latest kernel version by installing the **linux** package instead of **linux-lts** archlinux package. Archlinux wiki provides for info about this.

This are the quick steps to fix this: 
```
pacman -S linux-lts 
cd /etc/grub.d/
mv 06_OVH-kernel 12_OVH-kernel
cd /boot/grub
grub-mkconfig > grub.cfg
reboot
```
As soon as it reboots check the new kernel: 

```
# uname -a
Linux 4.1.18-1-lts #1 SMP Tue Feb 16 20:17:57 CET 2016 x86_64 GNU/Linux
```

And hopefully docker run with no issues. 
```
# docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world

03f4658f8b78: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:8be990ef2aeb16dbcb9271ddfe2610fa6658d13f6dfb8bc72074cc1ca36966a7
Status: Downloaded newer image for hello-world:latest

Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker Hub account:
 https://hub.docker.com

For more examples and ideas, visit:
 https://docs.docker.com/userguide/

```

Cheers!

**Comment by Klaim** [comment]
Yes that's basically what I did (see the links to the forum).

Thanks for the recap, I'll close this issue now.

**Comment by Flaburgan** [comment]
Hi guys. Commenting here in case it helps others:

I have the exact same problem, running on `3.14.32-xxxx-grs-ipv6-64` by OVH on a kimsufi, but I'm using Ubuntu 16.04 and not Arch, so this is related only to the kernel.

The issue is still present with docker 1.10.3 built with Go 1.6.1.

I would really like to see this fixed without having to use a custom kernel on the server. OVH strongly recommends to use the kernel they are providing. So, if anyone has another solution, or if this can be fixed docker side, it would be awesome. Thank you!

**Comment by jesusvazquez** [comment]
Hello @Flaburgan, I wouldn't use their kernel for many reasons: 
 - First because they can't mantain it as well as your distribution's developer comunity. 
 - It has problems with docker and virtual machines and also other weird behaviours that you can find in the net.

If you're a developer and you don't really want to care about the kernel but you want to use docker and vms I would install a kernel LTS recommended by your distro or I will rebuild the kimsufi server specifying that I don't want their kernel (They already have this option) and install the one you prefer from the list.

Cheers.

**Comment by Flaburgan** [comment]
The docker service was aimed to be a production installation. I installed it without docker. Thanks anyway.

**Comment by atay** [comment]
@jesusvazquez @Klaim
I have same issue, kimsufi server, cannot start docker, I want to update kernel, but not sure how. 
Do I need to install pacman or I can use some ready kernel? if so - which one? Any links, I'm sure it will be helpful not only for me, but more people coming here with same issue.

**Comment by jesusvazquez** [comment]
@atay if you checkout the comment I wrote the 6 of march there is a step by step guide.

**Comment by atay** [comment]
@jesusvazquez 
thanks, your comment was very helpful, but there is one thing missing there, how and where to download/install new kernel:
```
cd /tmp/
wget http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.4.13-xenial/linux-headers-4.4.13-040413-generic_4.4.13-040413.201606072354_amd64.deb
wget http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.4.13-xenial/linux-image-4.4.13-040413-generic_4.4.13-040413.201606072354_amd64.deb
wget http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.4.13-xenial/linux-headers-4.4.13-040413_4.4.13-040413.201606072354_all.deb
sudo dpkg -i *.deb
```

and then steps you mentioned:
```
cd /etc/grub.d/
mv 06_OVHKernel 12_OVHKernel
cd /boot/grub
grub-mkconfig > grub.cfg
reboot
```

and it works then. Thanks!

**Comment by jesusvazquez** [comment]
@atay when you install the linux-lts package it [downloads](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/linux-lts#n13) the kernel and place it in the right folder.

**Comment by dzackgarza** [comment]
Just wanted to add to the above comments that you may need to move (or remove) /boot/bzImage-3.14.32-xxxx-grs-ipv6-64, since grub-mkconfig still seems to pick it up before anything else.
